enhancements/baremetal/baremetal-config-raid-and-bios.md

```diff
@@ -0,0 +1,151 @@
+---
+title: Config-RAID-and-BIOS-for-Baremetal-IPI-deployments
+authors:
+  - "@fenggw-fnst"
+  - "@Hellcatlk"
+  - "@hs0210"
+  - "@zhouhao3"
+  - "@zhu1fei"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-05-18
+last-updated: 2021-05-25
+status: provisional
+see-also:
+  -
+replaces:
+  -
+superseded-by:
+  -
+---
+
+# Config RAID and BIOS for Baremetal IPI deployments
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement proposes to add the support of RAID and BIOS configuration for baremetal IPI deployments.
+
+## Motivation
+
+With the new features of [RAID configuration](https://github.com/metal3-io/baremetal-operator/pull/292) and [BIOS configuration](https://github.com/metal3-io/baremetal-operator/pull/302), openshift already has the ability to configure RAID and BIOS for baremetal.
+
+However, these features only work with a running cluster, users cannot configure desired RAID and BIOS during IPI deployments.
+
+### Goals
+
+- Allow users to manage the RAID and BIOS configuration for control plane hosts during installation.
+- IPI deployments can handle the configuration of RAID and BIOS.
+
+### Non-Goals
+
+- Add new or vendor specific options regarding RAID and BIOS.
+
+## Proposal
+
+The configuration of RAID and BIOS is finally handed over to Ironic during the IPI deployments.
+1. The user runs `create manifests` and then adds RAID and BIOS fields to the **BMH** created by the installer.
+2. The interface to terraform in the installer code parses the BIOS and RAID fields.
+3. [Terraform-provider-ironic](https://github.com/openshift-metal3/terraform-provider-ironic) gets the
+configuration of RAID and BIOS for control plane hosts and then passes it to Ironic([Baremetal Operator(BMO)](https://github.com/metal3-io/baremetal-operator)
+has supported the process of RAID and BIOS for worker nodes).
+
+### User Stories
+
+With the addition of this feature, the users can configure RAID and BIOS in IPI deployments.
+
+### Implementation Details/Notes/Constraints
+
+#### Add two fields
```

Comment:
I assume in the new version you don't add any new fields other than what has already been added to BMO?

Reply:
Yes. I have updated it.